### PR TITLE
viewer#2065 Region restart schedule floater

### PR DESCRIPTION
**Pull request by akleshchev**
image

### CODE REVIEW

**Review by marchcat — APPROVED** [review]
This looks awesome!

File: indra/newview/llfloaterregionrestartschedule.cpp [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+        LL_WARNS("Region") << "Saving region schedule, but RegionSchedule capability is not available" << LL_ENDL;
+    }
+}
+
+void LLFloaterRegionRestartSchedule::onCommitHours(const LLSD& value)
+{
+    S32 hours = value.asInteger();
+    if (mPMAMButton->getVisible())
+    {
+        llclamp(hours, 1, 12);
```
Comment (akleshchev) [inline indra/newview/llfloaterregionrestartschedule.cpp:199]: I'm not sure if I should translate 0 to 12 here.

**Review by maxim-productengine — APPROVED** [review]

**Comment by Ansariel** [comment]
My clock as 24 hours - I don't need an AM/PM selector!

**Comment by akleshchev** [comment]
> My clock has 24 hours - I don't need an AM/PM selector!

Mine too, but am/pm button is part of requested UI design and unfortunately viewer has no locale detection.

**Comment by Ansariel** [comment]
Time to file this as a bug because I as a user have no idea what AM/PM means! 😂

**Comment by akleshchev** [comment]
There is a debug setting to switch it

**Comment by marchcat** [comment]
BTW, what about the clock we have in the upper right corner of the main viewer window? I don't remember if it allows changing its time format, but if it does, that setting could also be used.

**Comment by Ansariel** [comment]
> BTW, what about the clock we have in the upper right corner of the main viewer window? I don't remember if it allows changing its time format, but if it does, that setting could also be used.

Not sure the LL viewer can, but Firestorm for sure can show 24h clock.

**Pull request merged**